03week/towersOfHanoi.js

```diff
@@ -37,25 +38,50 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece () {
-  // Your code here
-  if (n > 0) {
-    movePiece(n-1, a, c, b);
-    console.log("Move disk from " + a + " to " + c);
-    movePiece(n-1, b, a, c);
-  }
+function movePiece (startStack, endStack) {
+  // moves piece to endStack
+  //remove last number in startStack
+  //move it to the end of the endStack
+  stacks[endStack].push(stacks[startStack].pop());
 }
-movePiece(4, "A", "B", "C");
-
-function isLegal() {
-  // Your code here
 
-function checkForWin() {
-    // Your code here
+function isLegal(startStack, endStack) {
+  // confirms if move is legal
+  //startStack must have value to be moved.
+  //endStack must be empty or greater value than piece being moved
+  if (stacks[startStack].length === 0) {
+    return false;
+  } else if (stacks[endStack].length === 0) {
+    return true;
+  } else {
+    return stacks[startStack][stacks[startStack].length-1] < stacks[endStack][stacks[endStack].length-1];
+  }
+}
 
+function checkForWin(startStack, endStack) {
+  // tests for win after each move
+  //stack b or c could win if all 4 pieces moved successfully
+  if (stacks.b.length === 4 || stacks.c.length === 4) {
+    return true;
+  } else {
+    return false;
+  }
+}
 function towersOfHanoi(startStack, endStack) {
-  // Your code here
-
+  // legal move allows disk to be moved to endStack
+  const validEntry = (myStack) => {
```

Comment:
Nice data scrub but it doesn't appear that validEntry is being utilized in your app.